### PR TITLE
feat: sync label from node to globalnode

### LINKED ISSUE #550: feature: sync label  from node to globalnodes
**What would you like to be added**:
sync label  from node to globalnodes

**Why is this needed**:
Save the node label.
When a node is removed from the tenant control surface and then restored to the host cluster, the label needs to be restored.
Simultaneously update the state tag.

### PR DESCRIPTION
**Pull request by ONE7live**
#### What type of PR is this?
/kind feature

#### What does this PR do?

sync label from node to globalnode.
#### Which issue(s) does this PR fix?

Fixes #550
#### Special notes for your reviewer:

@OrangeBao 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go [hunk 1]
```diff
@@ -155,56 +155,44 @@ func (r *GlobalNodeController) SyncState(ctx context.Context, globalNode *v1alph
...
 
-		client = virtualClient
-	}
-
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		hostNode, err := client.CoreV1().Nodes().Get(ctx, globalNode.Name, metav1.GetOptions{})
-		if err != nil {
+		// If global node does not exist, create it
+		if _, err = r.KosmosClient.KosmosV1alpha1().GlobalNodes().Get(context.TODO(), globalNode.Name, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
```
Comment (OrangeBao) [inline pkg/kubenest/controller/global.node.controller/global_node_controller.go]: use ```ctx``` replace ```context.TODO```

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go [hunk 2]
```diff
@@ -155,56 +155,44 @@ func (r *GlobalNodeController) SyncState(ctx context.Context, globalNode *v1alph
...
+					}
+				}
+				globalNode = &v1alpha1.GlobalNode{
+					Spec: v1alpha1.GlobalNodeSpec{
+						NodeIP: nodeIP,
+						State:  v1alpha1.NodeReserved,
+					},
+				}
+				if _, err = r.KosmosClient.KosmosV1alpha1().GlobalNodes().Create(context.TODO(), globalNode, metav1.CreateOptions{}); err != nil {
+					klog.Errorf("global-node-controller: SyncState: can not create global node, err: %s", globalNode.Name)
```
Comment (OrangeBao) [inline pkg/kubenest/controller/global.node.controller/global_node_controller.go]: use ```ctx``` replace ```context.TODO```

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go [hunk 3]
```diff
@@ -155,56 +155,44 @@ func (r *GlobalNodeController) SyncState(ctx context.Context, globalNode *v1alph
...
-		if !needUpdate {
-			return nil
-		}
+		// Use management plane node label to override global node
+		updateGlobalNode := globalNode.DeepCopy()
+		updateGlobalNode.Labels = rootNode.Labels
 
-		if _, err := client.CoreV1().Nodes().Update(ctx, updateHostNode, metav1.UpdateOptions{}); err != nil {
-			klog.Errorf("global-node-controller: SyncState: update node label failed, err: %s", globalNode.Name)
+		if _, err = r.KosmosClient.KosmosV1alpha1().GlobalNodes().Update(context.TODO(), updateGlobalNode, metav1.UpdateOptions{}); err != nil {
```
Comment (OrangeBao) [inline pkg/kubenest/controller/global.node.controller/global_node_controller.go]: use ```ctx``` replace ```context.TODO```

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go [hunk 4]
```diff
@@ -155,56 +155,44 @@ func (r *GlobalNodeController) SyncState(ctx context.Context, globalNode *v1alph
...
+				}
+				globalNode = &v1alpha1.GlobalNode{
+					Spec: v1alpha1.GlobalNodeSpec{
+						NodeIP: nodeIP,
+						State:  v1alpha1.NodeReserved,
+					},
+				}
+				if _, err = r.KosmosClient.KosmosV1alpha1().GlobalNodes().Create(context.TODO(), globalNode, metav1.CreateOptions{}); err != nil {
+					klog.Errorf("global-node-controller: SyncState: can not create global node, err: %s", globalNode.Name)
+					return err
```
Comment (OrangeBao) [inline pkg/kubenest/controller/global.node.controller/global_node_controller.go]: Should end after creation

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go (hunk 1 above)
Comment (ONE7live) [reply pkg/kubenest/controller/global.node.controller/global_node_controller.go]: done

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go (hunk 2 above)
Comment (ONE7live) [reply pkg/kubenest/controller/global.node.controller/global_node_controller.go]: done

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go (hunk 3 above)
Comment (ONE7live) [reply pkg/kubenest/controller/global.node.controller/global_node_controller.go]: done

File: pkg/kubenest/controller/global.node.controller/global_node_controller.go (hunk 4 above)
Comment (ONE7live) [reply pkg/kubenest/controller/global.node.controller/global_node_controller.go]: ok, I've modified it

**Pull request merged**